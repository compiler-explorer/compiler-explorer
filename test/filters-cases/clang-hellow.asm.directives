.Lsection_info:
.Lsection_abbrev:
.Lsection_line:
.Lsection_str:
.Ldebug_range:
.Lsection_debug_loc:
.Ltext_begin:
main:                                   # @main
.Ltmp2:
.Lfunc_begin0:
# BB#0:
        pushq   %rbp
.Ltmp3:
.Ltmp4:
        movq    %rsp, %rbp
.Ltmp5:
.Ltmp6:
        movl    $.L.str, %edi
        xorb    %al, %al
        callq   printf
        movl    $str, %edi
        callq   puts
        xorl    %eax, %eax
        popq    %rbp
        ret
.Ltmp7:
.Ltmp8:
.Lfunc_end0:
.Ltmp9:
.Leh_func_end0:
.L.str:
        .asciz   "Hello world"
str:
        .asciz   "moo"
.Ltext_end:
.Ldata_end:
.Lsection_end1:
.Linfo_begin1:
.Linfo_end1:
.Labbrev_begin:
.Labbrev_end:
.Lset0 = .Lpubnames_end1-.Lpubnames_begin1 # Length of Public Names Info
.Lpubnames_begin1:
.Lset1 = .Linfo_end1-.Linfo_begin1      # Compilation Unit Length
.Lpubnames_end1:
.Lset2 = .Lpubtypes_end1-.Lpubtypes_begin1 # Length of Public Types Info
.Lpubtypes_begin1:
.Lset3 = .Linfo_end1-.Linfo_begin1      # Compilation Unit Length
.Lpubtypes_end1:
