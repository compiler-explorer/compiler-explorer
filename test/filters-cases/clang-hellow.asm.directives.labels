main:                                   # @main
# BB#0:
        pushq   %rbp
        movq    %rsp, %rbp
        movl    $.L.str, %edi
        xorb    %al, %al
        callq   printf
        movl    $str, %edi
        callq   puts
        xorl    %eax, %eax
        popq    %rbp
        ret
.L.str:
        .asciz   "Hello world"
str:
        .asciz   "moo"
