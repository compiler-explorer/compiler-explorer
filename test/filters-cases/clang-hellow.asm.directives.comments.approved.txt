{
  "asm": [
    {
      "labels": [],
      "source": null,
      "text": ".Lsection_info:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Lsection_abbrev:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Lsection_line:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Lsection_str:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ldebug_range:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Lsection_debug_loc:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ltext_begin:"
    },
    {
      "labels": [],
      "source": null,
      "text": "main:                                   # @main"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ltmp2:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Lfunc_begin0:"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 2
      },
      "text": "        pushq   %rbp"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ltmp3:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ltmp4:"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 2
      },
      "text": "        movq    %rsp, %rbp"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ltmp5:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ltmp6:"
    },
    {
      "labels": [],
      "source": {
        "column": 1,
        "file": null,
        "line": 3
      },
      "text": "        movl    $.L.str, %edi"
    },
    {
      "labels": [],
      "source": {
        "column": 1,
        "file": null,
        "line": 3
      },
      "text": "        xorb    %al, %al"
    },
    {
      "labels": [],
      "source": {
        "column": 1,
        "file": null,
        "line": 3
      },
      "text": "        callq   printf"
    },
    {
      "labels": [],
      "source": {
        "column": 3,
        "file": null,
        "line": 4
      },
      "text": "        movl    $str, %edi"
    },
    {
      "labels": [],
      "source": {
        "column": 3,
        "file": null,
        "line": 4
      },
      "text": "        callq   puts"
    },
    {
      "labels": [],
      "source": {
        "column": 3,
        "file": null,
        "line": 4
      },
      "text": "        xorl    %eax, %eax"
    },
    {
      "labels": [],
      "source": {
        "column": 1,
        "file": null,
        "line": 5
      },
      "text": "        popq    %rbp"
    },
    {
      "labels": [],
      "source": {
        "column": 1,
        "file": null,
        "line": 5
      },
      "text": "        ret"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ltmp7:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ltmp8:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Lfunc_end0:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ltmp9:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Leh_func_end0:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L.str:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .asciz   \"Hello world\""
    },
    {
      "labels": [],
      "source": null,
      "text": "str:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .asciz   \"moo\""
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ltext_end:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ldata_end:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Lsection_end1:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Linfo_begin1:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Linfo_end1:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Labbrev_begin:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Labbrev_end:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Lset0 = .Lpubnames_end1-.Lpubnames_begin1 # Length of Public Names Info"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Lpubnames_begin1:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Lset1 = .Linfo_end1-.Linfo_begin1      # Compilation Unit Length"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Lpubnames_end1:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Lset2 = .Lpubtypes_end1-.Lpubtypes_begin1 # Length of Public Types Info"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Lpubtypes_begin1:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Lset3 = .Linfo_end1-.Linfo_begin1      # Compilation Unit Length"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Lpubtypes_end1:"
    }
  ],
  "labelDefinitions": {
    ".L.str": 30,
    "main": 8,
    "str": 32
  }
}
